server/etcdserver/server.go

```diff
@@ -338,6 +338,15 @@ func NewServer(cfg config.ServerConfig) (srv *EtcdServer, err error) {
 			zap.Error(err),
 		)
 	}
+
+	if err = fileutil.RemoveTempFile(cfg.SnapDir()); err != nil {
```

Comment:
This function is simple enough, we can just place it somewhere in `etcdserver` package?

Reply:
IMHO etcdserver package is already too huge and should be candidate for splitting.